pkgs/development/haskell-modules/configuration-ghc-8.0.x.nix

```diff
@@ -38,6 +38,13 @@ self: super: {
   # cabal-install can use the native Cabal library.
   cabal-install = super.cabal-install.override { Cabal = null; };
 
+  idris = overrideCabal super.idris (drv: {
+    # "idris" binary cannot find Idris library otherwise while building.
+    # After installing it's completely fine though. Nix-specific issue so not
+    # reported.
+    preBuild = "export LD_LIBRARY_PATH=$PWD/dist/build:$LD_LIBRARY_PATH";
```

Comment:
Just to explain the issue: Idris consists of a Haskell library which constitutes the compiler a few small binaries that link to it (like the `idris` compiler driver). That stage succeeds. But when the build gets to the part where the `idris` binary will be invoked to start actually compiling the Idris stdlib, the build fails, since ld.so can't find the library against which `idris` is linked.

I get the feeling that if this were happening on more than just Nix, someone would have reported it, since it would mean the Idris build process was totally broken (i.e. because the default *is* to build a shared library, AFAIK), but I don't have an easy way to test. At any rate, without setting `LD_LIBRARY_PATH` like this, the build fails, and we also had to do the same thing for ghc 7.10.x, so is there a problem with fixing it for ghc 8.0.x as well?